filebeat/fileset/pipelines.go

```diff
@@ -48,6 +49,19 @@ type MultiplePipelineUnsupportedError struct {
 	minESVersionRequired common.Version
 }
 
+// Processor defines a single processors minimum version requirements.
+type Processor struct {
+	minVersion *common.Version
+	name       string
+	fn         func(pipelineID string, processor map[string]interface{}) error
```

Comment:
`fn` is not very descriptive. How about `MakeConfigCompatible`?

I think usage would be more clear if instead of passing in `pipelineID`, which is only used for logging, it accepts a `log *logp.Logger` parameter. Then you construct a logger that contains the contextual information.

`log := logp.NewLogger("fileset").With("pipeline", pipelineID)`

Then add more context when invoking each function `MakeConfigCompatible(config, log.With("processor_type", procType, "processor_index", i))`. Then any log statements should be clear as to exactly what processor they related.